specification/flag-evaluation/flag-evaluation.md

```diff
@@ -0,0 +1,195 @@
+# Flag Evaluation API
+
+## Overview
+
+The `evaluation API` allows for the retrieval of feature flag values, independent of any flag control plane or vendor. In the absence of a feature flag system implementation (a [provider](../provider/providers.md)) the `evaluation API` uses the "No-op provider", which simply returns the supplied default flag value.
+
+#### Initialization and Configuration
+
+##### Requirement 1.1
+
+> The `API`, and any state it maintains **SHOULD** exist as a global singleton, even in cases wherein multiple versions of the `API` are present at runtime.
+
+It's important that multiple instances of the `API` not be active, so that state stored therein, such as the registered `provider`, static global `evaluation context`, and globally configured `hooks` allow the `API` to behave predictably. This can be difficult in some runtimes or languages, but implementors should make their best effort that only a single instance of the `API` is used.
+
+##### Requirement 1.2
+
+> The `API` **MUST** provide a function to set the `provider`, which accepts an API-conformant `provider` implementation.
+
+```
+// example provider mutator
+OpenFeature.setProvider(new MyProvider());
+```
+
+see [provider](../provider//providers.md)
+
+##### Requirement 1.3
+
+> The `API` **MUST** provide a function for `hook` addition, which accepts an array of API-conformant `hooks`. This adds the specified hooks to any hooks already attached.
+
+```
+// example hook attachment
+OpenFeature.addHooks([new MyHook()]);
+```
+
+see [hooks](./hooks.md)
+
+##### Requirement 1.4
+
+> The API **MUST** provide a function for retrieving the `provider` implementation.
+
+```
+// example provider accessor
+OpenFeature.getProvider();
+```
+
+see [provider](../provider/provider.md)
+
+##### Requirement 1.5
+
+> The `API` **MUST** provide a function for creating a `client`, with an options object containing an optional field for the client `name`.
```

Comment:
Is an "options object" too prescriptive? One one hand, it seems less likely to cause breaking changes in many languages, since new properties can be added, on the other hand, it might not be idiomatic in some cases.

Reply:
if the `name` is optional, it does seem like the configuration itself should be an optional field. But if at a minimum it's required to pass a `name` then the options object makes sense being required.

Reply:
I'm OK with that. I think the only thing to consider is that in some languages it's less of a  breaking change to add new optional fields to an object, so things are slightly less brittle. I dont even know if that's true in Java though, I don't think so.

I will make this phrase less opinionated.

Reply:
> if the name is optional, it does seem like the configuration itself should be an optional field. But if at a minimum it's required to pass a name then the options object makes sense being required.

Ya, you're right, I guess this whole options field can be optional in languages that use an object.

Reply:
The wording seems too oriented around the languages you're familiar with and similar to the above comment about the array of hooks seems to me an implementation detail which could have different idioms in different languages. For example optional arguments are often keyword arguments in python.

```suggestion
> The `API` **MUST** provide a function for creating a `client` which accepts the following options:
>
> - name (optional): A logical string identifier for the client.
```

Reply:
Yep, this resolves @justinabrahms point too.